corehq/apps/api/resources/v0_5.py

```diff
@@ -1207,7 +1250,7 @@ def _query(cls, domain: str, local_timezone: pytz.tzinfo.DstTzInfo, params: dict
     @classmethod
     def _get_compound_filter(cls, key, params):
         compound_filter = Q()
-        for key, val in params.items():
+        for key, val in asdict(params)[key].items():
```

Comment:
You don't need to convert it to a dict for this, you can just do
```suggestion
        for key, val in getattr(params, key).items():
```
Also, I must've missed this in the first PR, but was the `key` passed in to this method just not used?

Reply:
[b8f4f9c](https://github.com/dimagi/commcare-hq/pull/33077/commits/b8f4f9c05fd2706256f4dd3308414827f6e9a048)

Oh ya I guess key wasn't used before